### PR TITLE
WI 1554 copy extensions

### PR DESCRIPTION
**Pull request by rooksdo**

### CODE REVIEW

File: TypeCobol/Parser.cs [hunk 1]
```diff
@@ -26,8 +26,8 @@ public class Parser
 		/// <summary>Optional custom symbol table to use for name and type resolution.</summary>
 		public SymbolTable CustomSymbols = null;
 
-		public string[] Extensions = { ".cbl", ".cpy" };
-		public string[] CopyExtensions = { ".cpy" };
+		public string[] Extensions = { ".cbl", ".cpy", ".copy" };
+		public string[] CopyExtensions = { ".cpy", ".copy" };
```
Comment (smedilol) [inline TypeCobol/Parser.cs]: Workspace.cs should also need to be modified.

I think we should have only one constant for copy extensions. Something like `DEFAULT_COPY_EXTENSIONS`
Maybe this constant could be in Helpers or in CompilationProject.

**Comment by rooksdo** [comment]
> smedilol said
> I'd like to make a performance test on our sample programs.
> At the beginning of the project, there was a performance issue with copy loading.

Here are the performance results after 3 consecutive runs for each branch v1.4 and v1.3.1 (average gain is 1,06% faster) :


| Test                                                  |                         | 1554 run_1 (ms) | 1554 run_2 (ms) | 1554 run_3 (ms) | v1.4 run_1 (ms) | v1.4 run_2 (ms) | v1.4 run_3 (ms) | Diff by mean (%) |
|-------------------------------------------------------|-------------------------|-----------------|-----------------|-----------------|-----------------|-----------------|-----------------|------------------|
| AntlrPerformanceProfiler                              |                         | 4436,00         | 4274,00         | 4261,00         | 4488,00         | 4439,00         | 4053,00         | -0,03            |
| Part1_Incremental_Cobol85_NoRedefines                 | First compilation       | 2330,00         | 3344,00         | 3439,00         | 3907,00         | 3768,00         | 3427,00         | -6,63            |
|                                                       | Incremental compilation | 355,88          | 505,33          | 492,93          | 564,43          | 500,73          | 491,70          | -0,68            |
| Part1_Incremental_TC_BigTypesNoProcedure              | First compilation       | 2130,00         | 3064,00         | 2971,00         | 3430,00         | 3443,00         | 3237,00         | -6,48            |
|                                                       | Incremental compilation | 483,28          | 541,18          | 537,70          | 554,25          | 584,40          | 534,70          | -0,37            |
| Part1_Incremental_TC_BigTypesWithProcedure            | First compilation       | 3298,00         | 3379,00         | 3357,00         | 3390,00         | 3154,00         | 3184,00         | 1,02             |
|                                                       | Incremental compilation | 700,85          | 653,45          | 660,98          | 588,80          | 616,55          | 610,20          | 0,67             |
| Part1_Incremental_TC_GlobaStorage                     | First compilation       | 3331,00         | 3432,00         | 3448,00         | 3121,00         | 3537,00         | 3373,00         | 0,60             |
|                                                       | Incremental compilation | 473,78          | 655,18          | 675,75          | 589,05          | 620,73          | 609,30          | -0,05            |
| Part1_FullParsing_Cobol85_NoRedefines                 |                         | 2233,95         | 3062,75         | 3052,55         | 2964,30         | 3045,25         | 2976,10         | -2,12            |
| Part1_FullParsing_TC_BigTypesNoProcedure              |                         | 2290,90         | 3249,60         | 3225,15         | 3080,30         | 3183,20         | 3002,65         | -1,67            |
| Part1_FullParsing_TC_BigTypesWithProcedure            |                         | 2289,10         | 3283,25         | 3294,00         | 3183,00         | 3234,25         | 3178,35         | -2,43            |
| Part1_FullParsing_TC_GlobalStorage                    |                         | 2296,65         | 3158,90         | 3327,70         | 2930,05         | 3110,40         | 3123,75         | -1,27            |
| Part2_Incremental_TC_UseALotOfTypes_001Time           | First compilation       | 588,00          | 1032,00         | 1045,00         | 917,00          | 931,00          | 998,00          | -0,60            |
|                                                       | Incremental compilation | 68,73           | 96,20           | 101,48          | 83,10           | 103,83          | 94,53           | -0,05            |
| Part2_Incremental_TC_UseALotOfTypes_100Times          | First compilation       | 695,00          | 1021,00         | 972,00          | 935,00          | 1000,00         | 1014,00         | -0,87            |
|                                                       | Incremental compilation | 113,60          | 166,53          | 172,65          | 135,15          | 162,40          | 167,98          | -0,04            |
| Part2_Incremental_TC_UseALotOfTypes_WithProc_100Times | First compilation       | 614,00          | 1019,00         | 1070,00         | 919,00          | 1026,00         | 1007,00         | -0,83            |
|                                                       | Incremental compilation | 114,33          | 164,75          | 169,90          | 137,00          | 163,90          | 171,75          | -0,08            |
| Part2_FullParsing_TC_UseALotOfTypes_001Time           |                         | 620,75          | 901,85          | 914,05          | 807,95          | 883,20          | 893,55          | -0,49            |
| Part2_FullParsing_TC_UseALotOfTypes_100Times          |                         | 660,70          | 1036,10         | 989,05          | 879,25          | 973,95          | 960,75          | -0,43            |
| Part2_FullParsing_TC_UseALotOfTypes_WithProc_100Times |                         | 693,00          | 994,00          | 990,60          | 891,75          | 980,40          | 975,85          | -0,57            |

**Comment by smedilol** [comment]
@rooksdo  There is no COPY in our performance tests ;)
But I've launched our private tests and it's a little faster.

File: TypeCobol/Parser.cs (hunk 1 above)
Comment (rooksdo) [reply TypeCobol/Parser.cs]: Done

File: CLI/test/ressources/extractUsedCopies_EI/input/Copies.rdz.tcbl [hunk 2]
```diff
@@ -5,9 +5,15 @@
...
-          01 PRTD20L. COPY  YPRTD20L.   
-          01 CONVEAL. COPY  YCONVEAL.
+          01 PRTD20L.  COPY  YPRTD20L.   
+          01 CONVEAL.  COPY  YCONVEAL.
           01 CONVEAL2. COPY  YCONVEAR.
+		  01 TRDF0L.   COPY  YTRDF0L.
+		  01 TRDF0LR.  COPY  YTRDF0LR.
+		  01 TRDF1L.   COPY  YTRDF1L.
+             COPY YTRDF2L  REPLACING ==:TRDF2L:== BY TDRF2L.
+             COPY YTRDF2LR REPLACING ==:TRDF2L:== BY TDRF2LR.
```
Comment (smedilol) [inline CLI/test/ressources/extractUsedCopies_EI/input/Copies.rdz.tcbl]: It's not related to this issue, but there is a bug here.

` COPY YTRDF2LR REPLACING ==:TRDF2L:== BY TDRF2LR.` should produce a copy missing for YTRDF2LR.
But here, it's considered as a COPY managed as a suffixed COPY (CPY).
Our EI rule is:
Copy used with replacing, consider it as a standard Cobol copy, and don't apply suffixing. Also search the file named YTRDF2LR.cpy / .copy.
Copy without replacing and starts with "Y". If the COPY has 8 chars: apply suffixing and search for the file name with the first 7 chars of the COPY.
But this file doesn't exists.

File: TypeCobol/Tools/CommandLine/TypeCobolCompiler.cs [hunk 3]
```diff
@@ -23,7 +23,7 @@ public static void Main(string[] args)
 
             string sourcePath = projectRootPath + @"TypeCobol.Test\Samples\EI Cobol samples\EI-Production";
             string[] programExtensions = { ".PGM" };
-            string[] copyExtensions = { ".CPY" };
+            string[] copyExtensions = { ".CPY", ".COPY" };
```
Comment (smedilol) [inline TypeCobol/Tools/CommandLine/TypeCobolCompiler.cs]: copyExtensions  can be replaced by DEFAULT_COPY_EXTENSIONS

File: TypeCobol.LanguageServer/Workspace.cs [hunk 4]
```diff
@@ -37,7 +37,7 @@ public class Workspace
         private SymbolTable _customSymbols;
         private string _rootDirectoryFullName;
         private string _workspaceName;
-        private string[] _extensions = { ".cbl", ".cpy" };
+        private string[] _extensions = { ".cbl", ".cpy", ".copy" };
```
Comment (smedilol) [inline TypeCobol.LanguageServer/Workspace.cs]: Can be replaced by Helpers.DEFAULT_EXTENSIONS

File: TypeCobol/Tools/CommandLine/TypeCobolCompiler.cs (hunk 3 above)
Comment (smedilol) [inline TypeCobol/Tools/CommandLine/TypeCobolCompiler.cs]: Can be replaced by Helpers.DEFAULT_COPY_EXTENSIONS

Comment (rooksdo) [reply TypeCobol/Tools/CommandLine/TypeCobolCompiler.cs]: done

File: TypeCobol.LanguageServer/Workspace.cs (hunk 4 above)
Comment (rooksdo) [reply TypeCobol.LanguageServer/Workspace.cs]: done.

File: TypeCobol/Tools/CommandLine/TypeCobolCompiler.cs (hunk 3 above)
Comment (rooksdo) [reply TypeCobol/Tools/CommandLine/TypeCobolCompiler.cs]: done.

**Pull request merged**